nsxt/resource_nsxt_policy_tier1_gateway_interface.go

```diff
@@ -67,24 +76,52 @@ func getGatewayInterfaceSubnetList(d *schema.ResourceData) []model.InterfaceSubn
 
 func resourceNsxtPolicyTier1GatewayInterfaceCreate(d *schema.ResourceData, m interface{}) error {
 	connector := getPolicyConnector(m)
-	client := locale_services.NewDefaultInterfacesClient(connector)
 
 	id := d.Get("nsx_id").(string)
 	tier1Path := d.Get("gateway_path").(string)
+	sitePath := d.Get("site_path").(string)
 	tier1ID := getPolicyIDFromPath(tier1Path)
-	localeService, err := getPolicyTier1GatewayLocaleServiceEntry(tier1ID, connector)
-	if err != nil {
-		return err
-	}
-	if localeService == nil {
-		return fmt.Errorf("Edge cluster is mandatory on gateway %s in order to create interfaces", tier1ID)
+	localeServiceID := ""
+	if isPolicyGlobalManager(m) {
+		localeServices, err := listPolicyTier1GatewayLocaleServices(connector, tier1ID, true)
+		if err != nil {
```

Comment:
Since model.LocaleServices is same for T0 and T1, could we perhaps reuse the code below for both T0 and T1?

Reply:
> Since model.LocaleServices is same for T0 and T1, could we perhaps reuse the code below for both T0 and T1?
Sure, I will wrap the common parts. Thanks.